### PR TITLE
feat(pipglines/pingcap/tidb/latest): enable bazel repo cache

### PR DESCRIPTION
**Pull request by wuhuizuo**
- add option in `$HOME/.bazelrc`
- mount the share pvc in pod

### CODE REVIEW

**Comment by wuhuizuo** [comment]
Replayed in https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_build/detail/ghpr_build/4890/pipeline/90

But: 💣  the option in make task cli args may make it always using the dir in image, it can not be updated, I need to make a PR in pingcap/tidb . @hawkingrei

File: pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml [hunk 1]
```diff
@@ -27,31 +27,25 @@ spec:
...
-
-                cpu_limit=$(cat /etc/containerinfo/cpu_limit);
-                mem_limit=$(cat /etc/containerinfo/mem_limit);
-                mem_limit=$(((mem_limit / 1048576) * 9 / 10 ));
-                :> /data/bazel; 
-                echo "build:ci --remote_cache=http://brc-bazel-remote.apps.svc --remote_timeout=10s" >> /data/bazel;
-                echo "build:ci --local_ram_resources=${mem_limit} --local_cpu_resources=${cpu_limit} -j ${cpu_limit}" >> /data/bazel;  
-                echo "test:ci --remote_cache=http://brc-bazel-remote.apps.svc --remote_timeout=10s" >> /data/bazel;
-                echo "test:ci --local_ram_resources=${mem_limit} --local_cpu_resources=${cpu_limit} -j ${cpu_limit}"  >> /data/bazel;
+              - /data/bazel-prepare-in-container.sh
```
Comment (hawkingrei) [inline pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml:48]: where is ```bazel-prepare-in-container.sh``` ?

File: pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml [hunk 2]
```diff
@@ -27,31 +27,25 @@ spec:
         - name: bazel-out-lower
           subPath: tidb/go1.19.2
           mountPath: /bazel-out-lower
-        - mountPath: /bazel-out-overlay
-          name: bazel-out-overlay
+        - name: bazel-out-overlay
+          mountPath: /bazel-out-overlay
+        - name: bazel-repository-cache
+          mountPath: /share/.cache/bazel-repository-cache
```
Comment (hawkingrei) [inline pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml:33]: Is it a physical disk ?

**Comment by hawkingrei** [comment]
Which PVC is it? We only need the physical disk PVC.

File: pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml (hunk 2 above)
Comment (wuhuizuo) [reply pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml:33]: pvc base on ceph-filesystem

File: pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml (hunk 1 above)
Comment (wuhuizuo) [reply pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml:48]: from the mounted k8s secret, source at: https://github.com/PingCAP-QE/ee-ops/blob/main/apps/prod/jenkins/post/_base/files/bazel-prepare-in-container.sh

**Comment by wuhuizuo** [comment]
> 



> Which PVC is it? We only need the physical disk PVC.

Ceph-filesystem disk can not make it global in k8s cluster for all building agents.

File: pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml (hunk 1 above)
Comment (wuhuizuo) [reply pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml:48]: same with current running, but only extracted and optimized the shells from k8s pod YAML.

**Comment by wuhuizuo** [comment]
https://github.com/PingCAP-QE/ci/pull/1517

> 

solved in: https://github.com/pingcap/tidb/pull/38750

**Comment by wuhuizuo** [comment]
replayed with https://github.com/pingcap/tidb/pull/38750: https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/ghpr_build/4965/

File: pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml (hunk 2 above)
Comment (hawkingrei) [reply pipelines/pingcap/tidb/latest/pod-ghpr_build.yaml:33]: Please do not use the ceph-filesystem.

**Pull request closed without merging**